src/pages/ImportData.js

```diff
@@ -30,36 +33,84 @@ const styles = {
+const ChoiceButton = ({ onClick, iconName, title, detail, tooltip }) => {
   return h(Clickable, {
+    onClick,
+    tooltip,
     style: {
-      ...style,
-      padding: '1rem',
+      padding: '1rem', marginTop: '1rem',
       display: 'flex', alignItems: 'center',
-      border: `1px solid ${colors.accent()}`, borderRadius: 4
+      border: `1px solid ${colors.accent(1)}`, borderRadius: 4
     },
-    hover: { backgroundColor: colors.accent(0.1) },
-    ...props
+    hover: { backgroundColor: colors.accent(0.1) }
   }, [
-    icon(iconName, { size: 24, style: { flex: 'none', marginRight: '1rem' } }),
+    icon(iconName, { size: 29, style: { flex: 'none', marginRight: '1rem', color: colors.accent(1) } }),
     div({ style: { flex: 1 } }, [
-      div({ style: { fontWeight: 'bold' } }, [title]),
+      div({ style: { fontWeight: 'bold', color: colors.accent(1) } }, [title]),
       div([detail])
     ]),
-    icon('angle-right', { size: 32, style: { flex: 'none', marginLeft: '1rem' } })
+    icon('angle-right', { size: 32, style: { flex: 'none', marginLeft: '1rem', color: colors.accent(1) } })
   ])
 }
 
 const ImportData = () => {
   const { workspaces, refresh: refreshWorkspaces } = useWorkspaces()
   const [isImporting, setIsImporting] = useState(false)
-  const { query: { url, format, ad, wid } } = Nav.useRoute()
+  const { query: { url, format, ad, wid, template } } = Nav.useRoute()
   const [mode, setMode] = useState(wid ? 'existing' : undefined)
   const [isCreateOpen, setIsCreateOpen] = useState(false)
+  const [isCloneOpen, setIsCloneOpen] = useState(false)
   const [selectedWorkspaceId, setSelectedWorkspaceId] = useState(wid)
+  const [selectedTemplateWorkspace, setSelectedTemplateWorkspace] = useState()
+  const [radioButtonSelected, setRadioButtonSelected] = useState()
+  const [templateWorkspaces, setTemplateWorkspaces] = useState()
+
+  const noteMessage = 'Note that the import process may take some time after you are redirected into your destination workspace.'
 
   const selectedWorkspace = _.find({ workspace: { workspaceId: selectedWorkspaceId } }, workspaces)
 
+  const getTemplates = fileContents => {
+    let temporaryList = []
+
+    !(_.isArray(template)) ? temporaryList = fileContents[template] :
+      _.forEach(currentTemplate => {
+        fileContents[currentTemplate] &&
+        temporaryList.push(fileContents[currentTemplate])
+      }, template)
+
+    return !temporaryList ? null : _.flatten(temporaryList)
+  }
+
+  Utils.useOnMount(() => {
+    const loadTemplateWorkspaces = async () => {
+      try {
+        const allTemplates = await fetch(`${getConfig().firecloudBucketRoot}/template-workspaces.json`).then(res => res.json())
+        setTemplateWorkspaces(getTemplates(allTemplates))
+      } catch (e) {
+        setTemplateWorkspaces(null)
+      }
+    }
+
+    loadTemplateWorkspaces()
+  })
+
+  const setSelectionInList = ws => {
+    setRadioButtonSelected(ws)
+    setSelectedTemplateWorkspace(ws)
```

Comment:
We can simplify the state handling here in a couple ways. This component already has the full list of workspaces from the `useWorkspaces` call above. If we use that, we can intersect that with the json list, so we only show workspaces that actually exist. We could also just store the selected id, and use that to look up the object, which minimizes our state. Also we probably don't need to make a separate details call, since we already have the necessary info.